### ISSUE TITLE
[a11y]: text area - Character limit information is not announced by the screen reader.

### ISSUE THREAD

**Issue by LyndseyBu** [carbon-design-system/carbon#19735]
### Package

@carbon/react

### Browser

Chrome, Firefox

### Operating System

Windows

### Package version

1.81

### React version

17.0.2

### Automated testing tool and ruleset

JAWS

### Assistive technology

JAWS

### Description

Issue description: Character limit information is not announced by the screen reader.

<img width="895" alt="Image" src="https://github.com/user-attachments/assets/a471dd32-ece1-4f15-b5c1-534bd3a34c3b" />

### WCAG 2.1 Violation

WCAG Mapping: 3.3.2 Labels or Instruction

### Reproduction/example

using storybook - https://react.carbondesignsystem.com/?path=/docs/components-textarea--overview

### Steps to reproduce

https://react.carbondesignsystem.com/?path=/docs/components-textarea--overview

Use JAWS and navigate to the textarea - the Charachter limit information is not announced by JAWS

<img width="1031" alt="Image" src="https://github.com/user-attachments/assets/3117c887-64e1-41d0-9d8b-a2bfaa74841a" />

### Suggested Severity

None

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [x] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems
